frontend/src/components/Workspace/FlowChart/FlowChartNode/MatlabFileNode.tsx

```diff
@@ -0,0 +1,328 @@
+import { NodeIdProps } from "store/slice/FlowElement/FlowElementType"
+import { setInputNodeFilePath } from "store/slice/InputNode/InputNodeActions"
+import {
+  selectInputNodeDefined,
+  selectInputNodeMatlabPath,
+  selectMatlabInputNodeSelectedFilePath,
+} from "store/slice/InputNode/InputNodeSelectors"
+import { setInputNodeMatlabPath } from "store/slice/InputNode/InputNodeSlice"
+import { FILE_TYPE_SET } from "store/slice/InputNode/InputNodeType"
+import { getMatlabTree } from "store/slice/Matlab/MatlabAction"
+import {
+  selectMatlabIsLoading,
+  selectMatlabNodes,
+} from "store/slice/Matlab/MatlabSelectors"
+import { MatlabTreeNodeType } from "store/slice/Matlab/MatlabType"
+import { selectCurrentWorkspaceId } from "store/slice/Workspace/WorkspaceSelector"
+import { AppDispatch } from "store/store"
+
+export const MatlabFileNode = memo(function MatlabFileNode(element: NodeProps) {
+  const defined = useSelector(selectInputNodeDefined(element.id))
+  if (defined) {
+    return <MatlabFileNodeImple {...element} />
+  } else {
+    return null
+  }
+})
+
+const MatlabFileNodeImple = memo(function MatlabFileNodeImple({
+  id: nodeId,
+  selected,
+}: NodeProps) {
+  const dispatch = useDispatch()
+  const filePath = useSelector(selectMatlabInputNodeSelectedFilePath(nodeId))
+  const onChangeFilePath = (path: string) => {
+    dispatch(setInputNodeFilePath({ nodeId, filePath: path }))
+  }
+
+  const onClickDeleteIcon = () => {
+    dispatch(deleteFlowNodeById(nodeId))
+  }
+
+  return (
+    <NodeContainer nodeId={nodeId} selected={selected}>
+      <button
+        className="flowbutton"
+        onClick={onClickDeleteIcon}
+        style={{ color: "black", position: "absolute", top: -10, right: 10 }}
+      >
+        ×
+      </button>
+      <FileSelect
+        nodeId={nodeId}
+        onChangeFilePath={(path) => {
+          if (!Array.isArray(path)) {
+            onChangeFilePath(path)
+          }
+        }}
+        fileType={FILE_TYPE_SET.MATLAB}
+        filePath={filePath ?? ""}
+      />
+      {filePath !== undefined && <ItemSelect nodeId={nodeId} />}
+      <Handle
+        type="source"
+        position={Position.Right}
+        id={toHandleId(nodeId, "matlab", "MatlabData")}
+        style={{ ...HANDLE_STYLE }}
+      />
+    </NodeContainer>
+  )
+})
+
+const ItemSelect = memo(function ItemSelect({ nodeId }: NodeIdProps) {
+  const dispatch = useDispatch<AppDispatch>()
+  const [open, setOpen] = useState(false)
+  const [fileSelect, setFileSelect] = useState("")
+
+  const structureFileName = useSelector(selectInputNodeMatlabPath(nodeId))
+
+  const onClickOk = () => {
+    dispatch(setInputNodeMatlabPath({ nodeId, path: fileSelect }))
+    setOpen(false)
+  }
+
+  const onClickCancel = () => {
+    setFileSelect("")
+    setOpen(false)
+  }
+
+  return (
+    <>
+      <Button variant="outlined" size="small" onClick={() => setOpen(true)}>
+        {"Structure"}
+      </Button>
```

Comment:
Apologies, this is an extra order.
Pls use icon button with following icon.
https://mui.com/material-ui/material-icons/?query=structure&selected=AccountTree
Tooltips text would be "Data structure".
And put the icon button next to upload via url button like CSV settings icon.

Apply same change to HDF5 node, too.

Reply:
@ReiHashimoto  i added it, pls check it again

Reply:
Ah, I mean here. Pls fix it.

![Screenshot 2023-12-26 at 13 44 22](https://github.com/arayabrain/barebone-studio/assets/42664619/cf495699-a78c-49fb-8a2c-751a3de53282)

Reply:
@ReiHashimoto i fixed it, pls check it again